AI/Building LLM system/logs-pillar.md

```diff
@@ -0,0 +1,136 @@
+---
+tags:
+  - llm
+  - observability
+  - log
+  - pillar
+title: Logs Pillar
+date: 2024-10-11
+description: 'Think of logs as a timeline of events in your app, recording every action or issue so you can easily track what’s happening behind the scenes.'
+authors:
+  - datnguyennnx
+---
+
+
+When you’re working with generative AI application, one thing that often gets overlooked is logging. Logging helps you keep track of what’s happening under the hood and gives you the insights you need to improve your model. Whether it's detecting errors or maintaining your AI runs smoothly, logging is fundamental. In this article, we'll look at why logging is important and how to use it to improve your LLM application.
+
+## Roles of logging in LLM application
+
+So, what’s logging? In simple terms, it’s about keeping a record of what happens between users and large language models (LLMs). This means saving both the questions users ask and the answers the model gives.
+
+If you look at the image below, it shows how an LLM app works. Logging is a key part of this because it captures things like the model’s inputs, outputs, the current state, memory being used, and the prompts running. This helps us see the big picture and keep track of how well the system is doing.
+
+![image.png](assets/logs-pillar-sample-rag-system.webp)
```

Comment:
I remember that providing an alt for an image won’t cause it to be rendered in a memo

```suggestion
![](assets/logs-pillar-sample-rag-system.webp)
```